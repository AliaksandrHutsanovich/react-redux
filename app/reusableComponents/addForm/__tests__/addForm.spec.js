import React from 'react';
import { act } from 'react-dom/test-utils';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Controller } from 'react-hook-form';
import AddForm, { AddForm as WithoutWrapper } from '../addForm';

const mockStore = configureStore();
const initialState = {};
const dispatch = jest.fn();
jest.mock('../../../sagas/execute', () => () => false);

describe('Unit tests of add form component', () => {
  const store = mockStore(initialState);
  store.dispatch = dispatch;
  const Component = shallow(<AddForm placeholder="Add new category" store={store} />);
  it('full render test', () => {
    expect(Component.shallow()).toMatchSnapshot();
  });

  const fragment = shallow(<WithoutWrapper placeholder="Add new category" dispatch={dispatch} />);
  const inputCategory = fragment.find(Controller).dive();
  it('input should be changable for adding new category', () => {
    inputCategory.simulate(
      'change',
      {
        target: {
          value: 'javascript',
        },
      },
    );
    expect(inputCategory.prop('value')).toEqual('javascript');
  });

  it('form should be sent to add new category', async () => {
    await act(async () => {
      fragment.find('form').prop('onSubmit')();
    });
    expect(dispatch).toHaveBeenCalled();
  });

  const withoutWrapper = shallow(<WithoutWrapper placeholder="Add new task" dispatch={dispatch} />);
  const inputTask = withoutWrapper.find(Controller).dive();
  it('input should be changable for adding new task', () => {
    inputTask.simulate(
      'change',
      {
        target: {
          value: 'javascript',
        },
      },
    );
    expect(inputTask.prop('value')).toEqual('javascript');
  });

  it('form should be sentable for adding new task', async () => {
    await act(async () => {
      withoutWrapper.find('form').prop('onSubmit')();
    });
    expect(dispatch).toHaveBeenCalled();
  });
});
