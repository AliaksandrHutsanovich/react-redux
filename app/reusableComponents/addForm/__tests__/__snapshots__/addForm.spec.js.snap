// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests of add form component full render test 1`] = `
<form
  autoComplete="off"
  className="formWrapper"
  name="addCategoryForm"
  onBlur={[Function]}
  onSubmit={[Function]}
>
  <Controller
    as={
      <Input
        className="input"
        placeholder="Add new category"
        type="text"
      />
    }
    control={
      Object {
        "defaultValuesRef": Object {
          "current": Object {},
        },
        "errorsRef": Object {
          "current": Object {},
        },
        "fieldArrayNamesRef": Object {
          "current": Set {},
        },
        "fieldsRef": Object {
          "current": Object {},
        },
        "formState": Object {
          "dirty": false,
          "dirtyFields": Set {},
          "isSubmitted": false,
          "isSubmitting": false,
          "isValid": false,
          "submitCount": 0,
          "touched": Object {},
        },
        "getValues": [Function],
        "isDirtyRef": Object {
          "current": false,
        },
        "mode": Object {
          "isOnBlur": false,
          "isOnSubmit": true,
        },
        "reValidateMode": Object {
          "isReValidateOnBlur": false,
          "isReValidateOnSubmit": false,
        },
        "readFormStateRef": Object {
          "current": Object {
            "dirty": false,
            "dirtyFields": false,
            "isSubmitted": true,
            "isSubmitting": false,
            "isValid": false,
            "submitCount": false,
            "touched": false,
          },
        },
        "register": [Function],
        "removeFieldEventListener": [Function],
        "resetFieldArrayFunctionRef": Object {
          "current": Object {},
        },
        "setValue": [Function],
        "touchedFieldsRef": Object {
          "current": Object {},
        },
        "triggerValidation": [Function],
        "unregister": [Function],
        "watchFieldArrayRef": Object {
          "current": Object {},
        },
      }
    }
    name="title"
    rules={
      Object {
        "required": "This is required",
      }
    }
  />
  <Button
    block={false}
    className="button"
    ghost={false}
    htmlType="submit"
    loading={false}
    type="primary"
  >
    Add
  </Button>
</form>
`;
